drivers/counter/counter_ll_stm32_rtc.c

```diff
@@ -648,6 +649,33 @@ static const struct rtc_stm32_config rtc_config = {
 	.pclken = rtc_clk,
 };
 
+#ifdef CONFIG_PM_DEVICE
+static int rtc_stm32_pm_action(const struct device *dev,
+			       enum pm_device_action action)
+{
+	const struct device *const clk = DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE);
+	const struct rtc_stm32_config *cfg = dev->config;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		if (clock_control_get_status(clk,
+					     (clock_control_subsys_t) &cfg->pclken[0])
+					     != CLOCK_CONTROL_STATUS_ON) {
```

Comment:
Open question:
Is there and actual gain to do this ?
What's the cost in reading status vs always performing clock_control_on ?

Reply:
You're right, it's not needed. And if we push PR support further for RTC counter, we'll disable the clock on suspend and reenable it anyway on resume.